Create Event Model

siddhant-0707 opened the pull request:
Created `Event` model as described in issue #14

shreyansh-nayak-ufl commented:
Check the `gorm.Model` object and attach it to the `Event` as that will be used to save these into the database.
Also, look into if you can move the `Event` into `models.go` file as sqlite package pulls from there to create migrations and tables.

siddhant-0707 commented:
I've introduced changes which should inculcate `gorm` into events. Also renamed the file to `models.go`. We will be keeping future models in this file.

siddhant-0707 commented:
@shreyansh-nayak-ufl I've made some changes to the `Event` model and have also created the `Bucket` model. If required, we can take a look into `Bucket` in another PR. I'll be creating an issue which will describe its use and structure.
